### PR TITLE
[full ci] Adding interface to mock system calls in unit tests

### LINKED ISSUE #2155: running "make tests" changes host systems state
eg; /etc/resolv.conf is removed.

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #2155

### CODE REVIEW

File: lib/etcconf/hosts.go [hunk 1]
```diff
@@ -68,6 +68,14 @@ func (w *hostsWalker) Next() string {
 	return s
 }
 
+func NewHosts(path string) Hosts {
+	if path == "" {
+		path = hostsPath
+	}
+
+	return &hosts{path: path, hosts: make(map[string]net.IP)}
```
Comment (matthewavery) [inline lib/etcconf/hosts.go]: Might be a little more reader friendly if there was a newline between variable assignments here.

**Comment by matthewavery** [comment]
other than my minor comment LGTM

**Comment by andrewtchin** [comment]
lgtm

**Comment by hmahmood** [comment]
♻️

**Pull request merged**